fibers/web/server.scm

```diff
@@ -103,19 +105,8 @@ on the procedure being called at any particular time."
            (extend-response response 'content-type
                             `(,@type (charset . ,charset))))
        (string->bytevector body charset))))
-   ((procedure? body)
-    (let* ((type (response-content-type response
-                                        '(text/plain)))
-           (declared-charset (assq-ref (cdr type) 'charset))
-           (charset (or declared-charset "utf-8")))
-      (sanitize-response
-       request
-       (if declared-charset
-           response
-           (extend-response response 'content-type
-                            `(,@type (charset . ,charset))))
-       (call-with-encoded-output-string charset body))))
-   ((not (bytevector? body))
+   ((not (or (bytevector? body)
+             (procedure? body)))
```

Comment:
IIRC,  the (web server) implementation didn't stream either, though that's a separate problem.